core/app/models/spree/order.rb

```
@@ -438,9 +438,11 @@ def ensure_shipping_address
     end
 
     def create_proposed_shipments
-      adjustments.shipping.delete_all
-      shipments.destroy_all
-      self.shipments = Spree::Stock::Coordinator.new(self).shipments
+      if !completed? && shipments.all?(&:pending?)
```

I think this is a sensible restriction, but I'd prefer this raised an exception rather than failing (doing nothing) silently.